src/_pytest/assertion/rewrite.py

```diff
@@ -1021,9 +1024,9 @@ def visit_Compare(self, comp: ast.Compare):
 def try_mkdir(cache_dir):
     """Attempts to create the given directory, returns True if successful"""
     try:
-        os.mkdir(cache_dir)
+        os.makedirs(fspath(cache_dir))
```

Comment:
`exist_ok=True` and then I think you can avoid the `except:` block maybe -- or just use `suppress`

Reply:
I changed to use `cache_dir.mkdir` as you suggest, unfortunately I could not get rid of `FileExistsError` handling (got some test failures), plus `test_try_mkdir` seems to be fine tuned already and I would have to change it now that `cache_dir` would be a `Path`, so I would prefer to not really change this if possible. 😬 

I also tried to use `suppress`, but it loses its appeal because we still need try/except for the other errors (so we can return `False` on some of them):

```diff
diff --git a/src/_pytest/assertion/rewrite.py b/src/_pytest/assertion/rewrite.py
index 415676aaa..0527a3639 100644
--- a/src/_pytest/assertion/rewrite.py
+++ b/src/_pytest/assertion/rewrite.py
@@ -1,5 +1,6 @@
 """Rewrite assertion AST to produce nice error messages"""
 import ast
+import contextlib
 import errno
 import functools
 import importlib.abc
@@ -1024,8 +1025,8 @@ warn_explicit(
 def try_mkdir(cache_dir):
     """Attempts to create the given directory, returns True if successful"""
     try:
-        os.makedirs(fspath(cache_dir))
-    except FileExistsError:
+        with contextlib.suppress(FileExistsError):
+            os.makedirs(fspath(cache_dir))
         # Either the pycache directory already exists (the
         # common case) or it's blocked by a non-dir node. In the
         # latter case, we'll ignore it in _write_pyc.
```

Reply:
I mean to use `makedirs(..., exist_ok=True)` ~essentially `mkdir -p` but only available in python3 :)